faker/providers/misc/__init__.py

```diff
@@ -91,7 +92,8 @@ def binary(cls, length=(1 * 1024 * 1024)):
 
         Default blob size is 1 Mb.
         """
-        return os.urandom(length)
+        blob = [random.randrange(256) for o in range(length)]
+        return bytes(blob) if sys.version_info[0] >= 3 else bytearray(blob)
```

Comment:
I see. Then I think we can just fix the test by checking for `isinstance(binary, (bytes, bytesarray))`